tests/garage/tf/regressors/test_bernoulli_mlp_regressor_with_model.py

```diff
@@ -0,0 +1,285 @@
+import pickle
+from unittest import mock
+
+from nose2.tools.params import params
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.optimizers import ConjugateGradientOptimizer, LbfgsOptimizer
+from garage.tf.regressors import BernoulliMLPRegressorWithModel
+from tests.fixtures import TfGraphTestCase
+
+
+def get_labels(input_shape, xs, output_dim):
+    if input_shape == (1, ):
+        label = [0, 0]
+        ys = 0 if np.sin(xs) <= 0 else 1
+        label[ys] = 1
+
+    elif input_shape == (2, ):
+        ys = int(np.round(xs[0])) ^ int(np.round(xs[1]))
+        if output_dim == 1:
+            label = ys
+        else:
+            label = [0, 0]
+            label[ys] = 1
+    return label
+
+
+def get_train_data(input_shape, output_dim):
+    if input_shape == (1, ):
+        data = np.linspace(-np.pi, np.pi, 1000)
+        obs = [{
+            'observations': [[x]],
+            'returns': [get_labels(input_shape, x, output_dim)]
+        } for x in data]
+
+    elif input_shape == (2, ):
+        # Generate 1000 points with coordinates in [0, 1]
+        x = np.linspace(0, 1, 100)
+        y = np.linspace(0, 1, 10)
+        data = np.dstack(np.meshgrid(x, y)).reshape(-1, 2)
+        obs = [{
+            'observations': [x],
+            'returns': [get_labels(input_shape, x, output_dim)]
+        } for x in data]
+    observations = np.concatenate([p['observations'] for p in obs])
+    returns = np.concatenate([p['returns'] for p in obs])
+    returns = returns.reshape((-1, output_dim))
+    return observations, returns
+
+
+def get_test_data(input_shape, output_dim):
+    if input_shape == (1, ):
+        paths = {
+            'observations': [[-np.pi / 2], [-np.pi / 3], [-np.pi / 4],
+                             [np.pi / 4], [np.pi / 3], [np.pi / 4]]
+        }
+        expected = [[1, 0], [1, 0], [1, 0], [0, 1], [0, 1], [0, 1]]
+
+    elif input_shape == (2, ):
+        paths = {'observations': [[0, 0], [0, 1], [1, 0], [1, 1]]}
+        if output_dim == 1:
+            expected = [[0], [1], [1], [0]]
+        else:
+            expected = [[1, 0], [0, 1], [0, 1], [1, 0]]
+    return paths, expected
+
+
+class TestBernoulliMLPRegressorWithModel(TfGraphTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.train_1dx2d = get_train_data((1, ), 2)
+        cls.train_2dx1d = get_train_data((2, ), 1)
+        cls.train_2dx2d = get_train_data((2, ), 2)
+
+        cls.test_1dx2d = get_test_data((1, ), 2)
+        cls.test_2dx1d = get_test_data((2, ), 1)
+        cls.test_2dx2d = get_test_data((2, ), 2)
+
+    @params(((1, ), 2), ((2, ), 1), ((2, ), 2))
+    def test_fit_normalized(self, input_shape, output_dim):
+        bmr = BernoulliMLPRegressorWithModel(
+            input_shape=input_shape, output_dim=output_dim)
+
+        if input_shape == (1, ):
```

Comment:
this is really gross. why not just call get_train_data???

Reply:
I agree. But I did that because that would repeatedly generate the same dataset. I'll undo this.